### PR TITLE
in peft finetune, only the trainable parameters need to be saved

### PR DESCRIPTION
**Pull request by sywangyi**
to reduce the storage size and also save the time of checkpoint saving while using deepspeed for training

# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Review by younesbelkada — APPROVED** [review]
LGTM, what do you think @pacman100 ?

**Comment by sywangyi** [comment]
@younesbelkada I have rebased to main. the code quality check failure is not related with the PR

**Comment by younesbelkada** [comment]
@sywangyi can you try to re-run `make fixup` with `pip uninstall black && pip install -U ruff==0.1.5` ?

**Review by younesbelkada — APPROVED** [review]
In principle this looks good! Would like to hear @pacman100 's thoughts before a core maintainer approval

**Comment by sywangyi** [comment]
@pacman100 could you help review the PR? Thanks.

**Review by pacman100 — APPROVED** [review]
Thank you @sywangyi for enabling only the trainable parameters to be saved during intermediate checkpointing when using PEFT with DeepSpeed, LGTM! 🚀

File: src/transformers/trainer.py
Comment (amyeroberts) [inline src/transformers/trainer.py:2425]: As this is repeated everywhere, let's abstract out this check to a module level function

```py
def _is_peft_model(model):
    return is_peft_available() and isinstance(self.model, PeftModel)
```


```suggestion
            if accept_exclude_frozen_parameters and _is_peft_model(model):
```

**Review by amyeroberts — APPROVED** [review]
Thanks for enabling this!

File: src/transformers/trainer.py
Comment (sywangyi) [reply src/transformers/trainer.py]: done

**Pull request merged**
